src/Symfony/Component/Messenger/Bridge/Amqp/Transport/Connection.php

```diff
@@ -534,11 +539,13 @@ public function exchange(): \AMQPExchange
         if (!isset($this->amqpExchange)) {
             $this->amqpExchange = $this->amqpFactory->createExchange($this->channel());
             $this->amqpExchange->setName($this->exchangeOptions['name']);
-            $this->amqpExchange->setType($this->exchangeOptions['type'] ?? \AMQP_EX_TYPE_FANOUT);
-            $this->amqpExchange->setFlags($this->exchangeOptions['flags'] ?? \AMQP_DURABLE);
+            if ('' !== $this->exchangeOptions['name']) {
```

Comment:
I'm not 100% sure about this part, since we are not calling `setType()` and `setFlags()` for the default exchange later on we would not be able to rely on `$this->amqpExchange->getType()` and `$this->amqpExchange->getFlags()`.
We could probably call `$this->amqpExchange->setType(\AMQP_EX_TYPE_DIRECT)` explicitly for the default exchange, but I'm not sure if it's needed, and also not sure about the `durable` flag (if it's always there for default exchange).

Reply:
Actually, it is only used in VarDumper at `src/Symfony/Component/VarDumper/Caster/AmqpCaster.php:136`, and it expects a valid exchange type, so probably calling `$this->amqpExchange->setType(\AMQP_EX_TYPE_DIRECT)` is a good idea